sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/client.py

```diff
@@ -66,75 +66,102 @@
      the client accepts or rejects the message. If set to `ReceiveAndDelete`, the service
      will assume successful receipt of the message and clear it from the queue. The
      default is `PeekLock`.
-    :type receive_settle_mode: ~uamqp.constants.ReceiverSettleMode
-    :param encoding: The encoding to use for parameters supplied as strings.
-     Default is 'UTF-8'
-    :type encoding: str
+    :paramtype receive_settle_mode: ~pyamqp.constants.ReceiverSettleMode
+    :keyword desired_capabilities: The extension capabilities desired from the peer endpoint.
+     To create an desired_capabilities object, please do as follows:
+        - 1. Create an array of desired capability symbols: `capabilities_symbol_array = [types.AMQPSymbol(string)]`
+        - 2. Transform the array to AMQPValue object: `utils.data_factory(types.AMQPArray(capabilities_symbol_array))`
+    :paramtype desired_capabilities: List
+    :keyword max_message_size: The maximum allowed message size negotiated for the Link.	
+    :paramtype max_message_size: int	
+    :keyword link_properties: Metadata to be sent in the Link ATTACH frame.	
+    :paramtype link_properties: dict	
+    :keyword prefetch: The receiver Link credit that determines how many	
+     messages the Link will attempt to handle per connection iteration.	
+     The default is 300.	
+    :paramtype prefetch: int
+    :keyword transport_type: The type of transport protocol that will be used for communicating with
+     the service. Default is `TransportType.Amqp` in which case port 5671 is used.
+     If the port 5671 is unavailable/blocked in the network environment, `TransportType.AmqpOverWebsocket` could
+     be used instead which uses port 443 for communication.
+    :paramtype transport_type: ~pyamqp.constants.TransportType
+    :keyword http_proxy: HTTP proxy settings. This must be a dictionary with the following
+     keys: `'proxy_hostname'` (str value) and `'proxy_port'` (int value).
+     Additionally the following keys may also be present: `'username', 'password'`.
+    :paramtype http_proxy: Dict
+    :keyword custom_endpoint_address: The custom endpoint address to use for establishing a connection to
+     the Event Hubs service, allowing network requests to be routed through any application gateways or
+     other paths needed for the host environment. Default is None.
+     The format would be like "sb://<custom_endpoint_hostname>:<custom_endpoint_port>".
+     If port is not specified in the `custom_endpoint_address`, by default port 443 will be used.
+    :paramtype custom_endpoint_address: str
+    :keyword connection_verify: Path to the custom CA_BUNDLE file of the SSL certificate which is used to
+     authenticate the identity of the connection endpoint.
+     Default is None in which case `certifi.where()` will be used.
+    :paramtype connection_verify: str
     """
 
-    def __init__(self, hostname, auth=None, **kwargs):
+    def __init__(self, hostname, **kwargs):
```

Comment:
feel free to ignore this comment, just an FYI 🙂 for Python 3 syntax you can also do:
```suggestion
    def __init__(self, hostname, *, auth=None, **kwargs):
      self._auth = auth
```